### ISSUE TITLE
Bootstrap 5 dropdown position relatively to body with boundary option

### ISSUE THREAD

**Issue by harnishdesign** [twbs/bootstrap#35774]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

I have a Bootstrap 5 dropdown menu inside owl-carousel. but dropdown is being cut off because has outer div overflow:hidden in owl-carousel.

[![enter image description here][2]][2]

So, I have try to position a dropdown relatively to body, not parent with boundary option from [Bootstrap Docs][1],

Bootstrap docs says that can only be done via javascript (not via data attributes), so I have try below two methods code. but no luck.

Where I'm going wrong. can you please suggest proper solution.

**1st try via JavaScript:**

    var dropdownElementList = [].slice.call(document.querySelectorAll('.dropdown-toggle'))
        var dropdownList = dropdownElementList.map(function (dropdownToggleEl) {
          return new bootstrap.Dropdown(dropdownToggleEl, {
            boundary: document.querySelector('#main-wrapper')
          })
        })

**2nd try Using function with popperConfig. Find from** [here][3]

    var dropdown = new bootstrap.Dropdown('.dropdown-toggle', {
      popperConfig: function (defaultBsPopperConfig) {
        
    return new bootstrap.Dropdown(dropdownTriggerEl, {
                popperConfig: {
                    strategy: "fixed"
                }
            });	
      }
    })

**Our HTML Code:**

    <div id="main-wrapper">
    <div class="owl-carousel owl-theme">
      <div class="item">
        <div class="d-flex align-items-center">
          <h4>Shilipp Sotocnik</h4>
          
          <div class="dropdown dropdown-lg d-inline-block ms-auto"> 
          <a href="#" role="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><i class="fas fa-ellipsis-h"></i></a>
            <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdownMenu1">
              <li><a class="dropdown-item" href="#">Link 1</a></li>
              <li><a class="dropdown-item" href="#">Link 2</a></li>
              <li><a class="dropdown-item" href="#">Link 3</a></li>
              <li><a class="dropdown-item" href="#">Link 4</a></li>
            </ul>
          </div>
          
        </div>
      </div>
    </div>
    </div>

  [1]: https://getbootstrap.com/docs/5.1/components/dropdowns/#options
  [2]: https://i.stack.imgur.com/LGAs1.jpg
  [3]: https://stackoverflow.com/a/65259888/1923139

### Reduced test cases

cxx

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Firefox

### What version of Bootstrap are you using?

v5.1.0

**Comment by GeoSot** [comment]
Hey @harnishdesign .  Can you please include and a codepen demo?

**Comment by harnishdesign** [comment]
> Hey @harnishdesign . Can you please include and a codepen demo?

Please check it out here: https://jsfiddle.net/harnishdesign/obg7favd/9/

**Comment by GeoSot** [comment]
Did some testing there and found out that `position:fixed` at least is set properly. But seems that it is not enough to bypass owl `overflow:hidden` :/

![image](https://user-images.githubusercontent.com/22406063/152328390-25193195-d067-42dd-91a1-a14a702c4a83.png)

**Comment by harnishdesign** [comment]
@GeoSot Yes, i know. So, I want to dropdown position relatively to body instead of parent element. and,
It works to with v4.5 with html attribute `data-boundary="window"`. But I couldn't make it work with v5. The [Bootstrap 5 documentation](https://getbootstrap.com/docs/5.1/components/dropdowns/#options) says that can only be done via javascript (not via data attributes), so I have try some javascript (as mention in question). but, no luck. So, I have to looking for correct javascript.

**Comment by FirstVertex** [comment]
BUMP!

Append dropdown to Body is common in Bootstrap pre version 5.  Why does TWBS think that all of a sudden in Bootstrap 5, nobody needs that feature anymore?  The doc is completely void of any instruction how to place dropdown as a child of Body.  This is not some optional nice-to-have feature, it is essential for properly using Bootstrap dropdown when combined with other elements who have overflow: hidden.

**Comment by FirstVertex** [comment]
I want to know also, how to append Bootstrap Dropdown to body.  It was standard concept for all previous versions of Bootstrap.  This is a blocker for our application to move to Bootstrap 5, as we have conditions where a Popover must have a Dropdown.

Here is my post to Stack Overflow asking for help
https://stackoverflow.com/questions/72031591/popper-js-2-append-to-body

What's puzzling is Popper.js v2 doc has some tiny mention about it's possible to append to body, but there is no example anywhere how to do that.
https://popper.js.org/docs/v2/performance/#attaching-elements-to-the-dom

I discovered ngBootstrap implementation which makes it's own createPopper() implementation.
https://github.com/ng-bootstrap/ng-bootstrap/blob/1ab5d2113007c52f7832b0898abc54b5a684bdc4/src/util/positioning.ts
this seems like a huge amount of effort to get the Dropdown into the body.  In all previous Bootstrap versions it was just an input switch.  position: fixed strategy has lots of problems and is not a substitute for append to body.

**Comment by jfmdev** [comment]
I'm having the same problem, I'm doing: 

```js
import { Dropdown } from "bootstrap";
...
new Dropdown(document.getElementById('my-dropdown'), { 
  boundary: document.querySelector('body')
});

```

But the `boundary` parameter is completely ignored.

**Comment by nicped** [comment]
Ran into the same issue and by adding `position-fixed` to the `dropdown-menu` element, the dropdown does not get hidden by its scrolling parent

https://codepen.io/nicped/pen/oNEVVNw

**Comment by sairexes** [comment]
> Ran into the same issue and by adding `position-fixed` to the `dropdown-menu` element, the dropdown does not get hidden by its scrolling parent
> 
> https://codepen.io/nicped/pen/oNEVVNw

This works! thank you!

**Comment by jfmdev** [comment]
If adding `position-fixed` don't works, you can use the `strategy: "fixed"` option, as suggested on https://github.com/twbs/bootstrap/issues/36358#issuecomment-1162479957

```js
new Dropdown(document.getElementById('my-dropdown'), { 
  boundary: document.querySelector('body'),
  popperConfig: function (defaultBsPopperConfig) {
            return { ...defaultBsPopperConfig, strategy: "fixed"};
        }
});
```

**Comment by GreathostRo** [comment]
After more than a few hours and various attempts I found the working version:

```
const dropdowns = document.querySelectorAll('.dropdown-toggle-simple')
const dropdown = [...dropdowns].map((dropdownToggleEl) => new bootstrap.Dropdown(dropdownToggleEl, {
    popperConfig(defaultBsPopperConfig) {
        return { ...defaultBsPopperConfig, strategy: 'fixed' };
    }
}));
```

**Comment by ahmeti** [comment]
> Ran into the same issue and by adding `position-fixed` to the `dropdown-menu` element, the dropdown does not get hidden by its scrolling parent
> 
> https://codepen.io/nicped/pen/oNEVVNw

Thanks for the trick... The "position-fixed" class you specified makes an incorrect positioning in table. I recommend using the "position-absolute" class as a workaround.

**Comment by gsartori** [comment]
Same problem here, we have dropdowns into table rows. How can we avoid the table clipping? Usong "position-fixed" or "position-absolute" is not helping. We are on Bootstrap 5.3 with popper 2.11.6 using data-bs-boundary="body" in a button with class="dropdown-toggle".

What's the correct way to have dropdowns in a table row?

**Comment by nicped** [comment]
> What's the correct way to have dropdowns in a table row?

Try to create a pin with the issue - then maybe I can come up with a solution

**Comment by Stanley-shrimp** [comment]
> Hey [@harnishdesign](https://github.com/harnishdesign) . Can you please include and a codepen demo?





https://stylesheets.space/questions/place-the-image-at-the-bottom-of-its-parent-div-without-using-absolute-positioning

**Comment by mdo** [comment]
Menu on Floating UI supports `container` (render the menu elsewhere, e.g. on `<body>`), `boundary`, and `strategy: 'fixed'`, which covers positioning out of clipping/overflow ancestors. Closing — please open a new issue if these don't cover your case in v6-dev.
